File: src/dataset/c/collatz-conjecture_exercism-yorange1/claude/ClaudeSonnet4_collatz_conjecture_v4.c
#include "collatz_conjecture.h"

#define is_even(n) ((n & 1) == 0)

int steps(int start)
{
    if (start <= 0)
    {
        return ERROR_VALUE;
    }

    int count = 0;
    unsigned int n = (unsigned int)start;

    while (n != 1)
    {
        if (is_even(n))
        {
            n >>= 1;
            count++;
        }
        else
        {
            n = n * 3 + 1;
            count++;
            if (is_even(n))
            {
                n >>= 1;
                count++;
            }
        }
    }

    return count;
}

File: src/dataset/python/collatz-conjecture_exercism-fruit-in/claude/ClaudeSonnet4_collatz_conjecture_v4.py
def steps(number):
    if number <= 0:
        raise ValueError(r".+")
    
    count = 0
    while number != 1:
        if number & 1:
            number = 3 * number + 1
        else:
            number >>= 1
        count += 1
    
    return count